index.js

```diff
@@ -38,6 +43,10 @@ function prepareAndSendToken (req, res, type, token, options, cb) {
   var params = {};
   params[type] = null;
   params.RelayState = req.body.RelayState || req.query.RelayState || options.relayState || '';
+  if (params.RelayState === '') {
+    // if there is no RelayState value, the parameter should be omitted from the signature computation
```

Comment:
> the parameter should be omitted from the signature computation

please note that RelayState parameter isn't involved in a signature computation for `HTTP-POST` and `HTTP-Redirect without deflate encoding` (just for `HTTP-Redirect with deflate encoding`)

Reply:
Hello @sebadoom , 

> So this means that the old behavior was also incorrect in the sense that it was using the RelayState (if passed in) as part of the signature computation for all cases (as long as RelayState was a non-empty string). Did I understand that correctly?

Actually no, if you look closely, in the case of HTTP Post or Redirect without deflate, only the "token" (a LogoutRequest) is signed, which does not contain RelayState. 

> This would also mean that the behavior in this PR is also incorrect because it uses RelayState for the signature in cases other than HTTP-Redirect + deflate (as long as the RelayState is a non-empty string).

Also no, your change just does not send the RelayState as a paremeter, but the signature is anyway unchanged. 

> Should I go ahead and change the behavior in this library so that RelayState is only used as part of the signature when HTTP-Redirect + deflate is used? With regards to the workaround introduced for Azure AD, I starting to get more inclined to just change the behavior of SAML logout to always send a RelayState (though this would be perhaps an unexpected change for other IdPs, I need to validate with Product). Thoughts? //cc: @julienwoll

This is the behavior today basically. We approved this workaround for Azure, but the ideal solution for me would be to file a ticket on Azure side and see their answer. Unfortunately we cannot make a Azure only fix here. 

I would suggest then: 
- File ticket to Azure 
- Go ahead and merge this one and check if all is fine in server 
- Release and let's keep a look on the list of customers I shared in the ESD 

Works for you ?

Reply:
This works for me, thanks. I am just concerned about adding workaround for specific providers in this library.

The fix I was proposing in my previous comments was to have auth0-server send a non-empty RelayState for _all_ logouts (Azure AD problem triggers when we send an empty string RelayState). This would however be a more invasive change and may trigger unexpected behavior in other IdPs. I'll merge this tomorrow morning and also send the PR for auth0.server.

Thanks!